### DISCUSSION TITLE
maui obfuscator

### DISCUSSION THREAD

**Discussion by tjiakwokyung28** [dotnet/maui#20662] [General]
what is the best obfucation for net maui? open source for net maui obfuscation?

**Comment by developer9969** [comment]
Just a quick one the cheap ones don't work and rubbish and the good one you need at least 8 to 10  thousands pounds and if have to take the app for pen testing good luck

**Reply by tjiakwokyung28** [reply]
maui is not rubbish, compared to flutter, maui is much better, there are many things that can't be made with flutter but can be made with maui. maui is the future programming

**Reply by developer9969** [reply]
Hey - I think you misread my comment or my comment are not clear. Let me clarify I am in the same boat as you on this one and I have evaluated quite a few obfuscator tools (third parties)

1. Maui like Xamarin does not offer any obfuscation out of the box!
2. You got to purchase one, 
3. The third party ones come at all sorts of prices, and my experience was that many just dont work!

your options I am afraid is to purchase one as Microsoft will never ever build one. Hope this clarifies for you and is nothing to do with Maui or Flutter, in other languages they have one that comes out of the box.

**Comment by williamchenicon** [comment]
Obfuscar is free and works OK but does not obfuscate XAML and view models. This limitation is manageable if you are not using MVVM.
